lib/Reader/readerir.cpp

```diff
@@ -5015,14 +5264,9 @@ IRNode *GenIR::genNullCheck(IRNode *Node) {
   return Node;
 }
 
-// Generate array bounds check.
-IRNode *GenIR::genBoundsCheck(IRNode *Array, IRNode *Index) {
+void GenIR::genBoundsCheck(Value *ArrayLength, Value *Index) {
```

Comment:
The old model of returning a "new" array here was there to try and enforce dependence of subsequent array users on the check. Because we currently create explicit checks and control flow we have control dependence. If we go to an implicit check model we may lose that and have to fall back to data dependence.